src/goto-programs/goto_function.h

```diff
@@ -29,33 +29,35 @@ class goto_functiont
   goto_programt body;
 
   /// The type of the function, indicating the return type and parameter types
+  /// This is deprecated; get the type from the namespace instead.
   code_typet type;
 
   typedef std::vector<irep_idt> parameter_identifierst;
 
   /// The identifiers of the parameters of this function
   ///
-  /// Note: This variable is currently unused and the vector is thus always
-  /// empty. In the future the code base may be refactored to fill in the
-  /// parameter identifiers here when creating a `goto_functiont`. For now the
-  /// parameter identifiers should be retrieved from the type (`code_typet`).
+  /// Note: This is now the preferred way of getting the identifiers of the
+  /// parameters. The identifiers in the type will go away.
   parameter_identifierst parameter_identifiers;
 
   bool body_available() const
   {
     return !body.instructions.empty();
   }
 
+  /// This is deprecated; get the type from the namespace instead.
```

Comment:
We have a DEPRECATED ([`util/deprecate.h`](https://github.com/diffblue/cbmc/blob/develop/src/util/deprecate.h)) macro for this purpose, and doxygen has the special [`\deprecated`](http://www.doxygen.nl/manual/commands.html#cmddeprecated) annotation for documentation

Reply:
Done.